File: Selenium/Activities/Activity17.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

import java.util.List;

public class Activity17 {
        public static void main(String[] args) {
            // Set the path to the ChromeDriver executable
            WebDriverManager.chromedriver().setup();
            ChromeDriver driver= new ChromeDriver();
            driver.get("https://v1.training-support.net/selenium/tables");

            // Get the title of the page and print it to the console
            String pageTitle = driver.getTitle();
            System.out.println("Page Title: " + pageTitle);

            // Locate the second table using XPath
            WebElement table = driver.findElement(By.xpath("//table[@id='sortableTable']"));

            // Find the number of rows and columns in the table and print them
            List<WebElement> rows = table.findElements(By.xpath(".//tbody/tr"));
            int numRows = rows.size();
            int numColumns = table.findElements(By.xpath(".//thead/tr/th")).size();
            System.out.println("Number of Rows: " + numRows);
            System.out.println("Number of Columns: " + numColumns);

            // Find and print the cell value at the second row second column
            WebElement cellValue = rows.get(1).findElement(By.xpath(".//td[2]"));
            System.out.println("Cell Value at 2nd row 2nd column: " + cellValue.getText());

            // Click the header of the first column to sort by name
            WebElement nameHeader = table.findElement(By.xpath(".//th[1]"));
            nameHeader.click();

            // Find and print the cell value at the second row second column again
            cellValue = rows.get(1).findElement(By.xpath(".//td[2]"));
            System.out.println("Cell Value at 2nd row 2nd column after sorting: " + cellValue.getText());

            // Print the cell values of the table footer
            WebElement tableFooter = driver.findElement(By.xpath("//tfoot"));
            List<WebElement> footerCells = tableFooter.findElements(By.xpath(".//td"));
            for (WebElement footerCell : footerCells) {
                System.out.println("Footer Cell Value: " + footerCell.getText());
            }

            // Close the browser
            driver.close();
        }
    }

File: Selenium/Activities/Activity17.py
from  selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service as ChromeService
from selenium.webdriver.support.select import Select
from webdriver_manager.chrome import ChromeDriverManager
driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))


# Open the webpage
driver.get("https://v1.training-support.net/selenium/selects")

# Get the title of the page and print it
pageTitle = driver.title
print("Page Title:", pageTitle)

# Find the dropdown
single_select = driver.find_element(By.ID, "single-select")
# Pass the dropdown WebElement to the Select class and initialize the select object
dropdown = Select(single_select)

# Select the second option using the visible text.
dropdown.select_by_visible_text("Option 2")
# Print the selected option using the first_selected_option method
# first_selected_option returns a WebElement.
# Use the text method to get the name of the selected option
print("Selected option: ", dropdown.first_selected_option.text)

# Select the third option using the index.
dropdown.select_by_index(3)
print("Selected option: ", dropdown.first_selected_option.text)

# Select the fourth option using the value.
dropdown.select_by_value("4")
print("Selected option: ", dropdown.first_selected_option.text)

# Get all the options and print them to the console.
select_options = dropdown.options
for option in select_options:
    print("Option: ", option.text)